### PR TITLE
#256 Michelson integration - errors fixed

### PR DESCRIPTION
**Pull request by dorians**

### CODE REVIEW

File: src/main/scala/tech/cryptonomic/conseil/tezos/michelson/dto/MichelsonExpression.scala [hunk 1]
```diff
@@ -25,7 +25,7 @@ sealed trait MichelsonExpression extends MichelsonElement
 case class MichelsonType(prim: String, args: List[MichelsonExpression] = List.empty) extends MichelsonExpression
 
 /* Class representing an int constant */
-case class MichelsonIntConstant(int: Int) extends MichelsonExpression
+case class MichelsonIntConstant(int: Long) extends MichelsonExpression
```
Comment (piotrkosecki) [inline src/main/scala/tech/cryptonomic/conseil/tezos/michelson/dto/MichelsonExpression.scala]: just a thought about naming, you changed `Int` to `Long`, but name of the case class stayed the same, maybe it is worth to change it to something that does not imply specific scala type like that, to something like `MichelsonNumericConstant`, `MichelsonIntegerConstant`? WDYT?

Comment (dorians) [reply src/main/scala/tech/cryptonomic/conseil/tezos/michelson/dto/MichelsonExpression.scala]: The problem is that `int` is the name of a field which contains long value encodes as a string ;)
That's what it looks like in Michelson and I agree that it's confusing. I'll look at this again tomorrow with a fresh head.

Comment (piotrkosecki) [reply src/main/scala/tech/cryptonomic/conseil/tezos/michelson/dto/MichelsonExpression.scala]: So in this case it makes more sense 👍

File: src/main/scala/tech/cryptonomic/conseil/tezos/michelson/renderer/MichelsonRenderer.scala [hunk 2]
```diff
@@ -7,15 +7,20 @@ object MichelsonRenderer {
...
+      case MichelsonComplexInstruction(prim, args) => s"$prim ${args.map(_.render()).mkString(" ")}"
       case MichelsonInstructionSequence(args) => s"{ ${args.map(_.render()).mkString(" ; ")} }"
+      case MichelsonEmptyInstruction => "{}"
 
+      // expressions
       case MichelsonType(name, List()) => name
       case MichelsonType(name, args) => s"($name ${args.map(_.render()).mkString(" ")})"
       case MichelsonIntConstant(constant) => constant.toString
-      case MichelsonStringConstant(constant) => constant
+      case MichelsonStringConstant(constant) => "\"%s\"".format(constant)
```
Comment (ivanopagano) [inline src/main/scala/tech/cryptonomic/conseil/tezos/michelson/renderer/MichelsonRenderer.scala]: You can alternatively use "raw strings" when you need to embed quoted values, as in
`s""""$constant""""`

It's debatable which is more easy to read, yet I argue that the latter is harder to write incorrectly, because the compiler might catch non-matching quotes, or an incorrect interpolation...
With the `"...".format()` version, you'll find most errors at runtime

File: src/test/scala/tech/cryptonomic/conseil/tezos/michelson/parser/JsonParserSpec.scala [hunk 3]
```diff
@@ -90,8 +90,8 @@ class JsonParserSpec extends FlatSpec with Matchers {
 
     parse[MichelsonInstruction](json) should equal(Right(
       MichelsonInstructionSequence(List(
-        MichelsonComplexInstruction("DIP", MichelsonInstructionSequence(List(
-          MichelsonSimpleInstruction("DUP"))))))))
+        MichelsonComplexInstruction("DIP", List(MichelsonInstructionSequence(List(
```
Comment (ivanopagano) [inline src/test/scala/tech/cryptonomic/conseil/tezos/michelson/parser/JsonParserSpec.scala]: Trivial comment: using indentation and formatting might make it easier to read such nesting of Lists.

I don't know if using the `a :: b :: Nil` format might make it all more parseable to the reader

File: src/test/scala/tech/cryptonomic/conseil/tezos/michelson/parser/JsonParserSpec.scala [hunk 4]
```diff
@@ -90,8 +90,8 @@ class JsonParserSpec extends FlatSpec with Matchers {
 
     parse[MichelsonInstruction](json) should equal(Right(
```
Comment (ivanopagano) [inline src/test/scala/tech/cryptonomic/conseil/tezos/michelson/parser/JsonParserSpec.scala]: If you prefer, scalatest will provide with combinators to extract `Right` using a DSL

```scala
parse[MichelsonInstruction](json).right should equal (...)
```

File: src/test/scala/tech/cryptonomic/conseil/tezos/michelson/parser/JsonParserSpec.scala (hunk 3 above)
Comment (ivanopagano) [reply src/test/scala/tech/cryptonomic/conseil/tezos/michelson/parser/JsonParserSpec.scala]: The alternatives would look something like
```scala
MichelsonInstructionSequence(
  List(
    MichelsonComplexInstruction("DIP", 
      List(
        MichelsonInstructionSequence(
          List(
            MichelsonSimpleInstruction("DUP")
          )
        )
      )
    )
  )
)
//vs.
MichelsonInstructionSequence(
  MichelsonComplexInstruction("DIP",
    MichelsonInstructionSequence(
      MichelsonSimpleInstruction("DUP") :: Nil
    ) :: Nil
  ) :: Nil
)
```

Comment (dorians) [reply src/test/scala/tech/cryptonomic/conseil/tezos/michelson/parser/JsonParserSpec.scala]: I'm refactoring this structure now to be able to add support for LAMBDA instruction and I'll need to change tests anyway so I'll try with ``a :: b :: Nil`` syntax. It might look better indeed.
